### ISSUE TITLE
Add example for Function Call with streaming 

### ISSUE THREAD

**Issue by 64bit** [64bit/async-openai#85]
Just like `examples/function-call` but using streaming call. 

Note: There is open PR at the moment but no way to test it #83

**Comment by victor-wei126** [comment]
@64bit going to try to tackle this issue first.

I've been skimming the codebase and also the OpenAPI spec. Just curious (and this might help me understand the codebase better) - what was the main design and considerations you took while developing this library to use the OpenAI API? Did you mostly use the OpenAPI spec or look at the design of their official python library?

**Comment by 64bit** [comment]
The primary source was OpenAPI spec and async-stripe was source of inspiration, you might find evolution of this library if you look at early commits. When I started I couldnt find good resources to generate code from spec, and json spec is not always up to date, sometimes has human errors. Moreover there are complex situations like streaming where an input field like `"stream": true` determines SSE response. 

So I decided to build this library with hand instead of relying on auto generation (although auto generation would be ideal for Rust in long term) because my focus was on correctness, auto retries on rate limit and full API coverage - all the things to make library production ready.

I had skimmed through the offcial python library too, and then went with spec as source of truth and grouping ("images", "fine-tunes" etc.) based on OpenAPI spec + RESTful spec.
